cpp/src/arrow/compute/api_scalar.cc

```diff
@@ -240,6 +243,15 @@ StrptimeOptions::StrptimeOptions(std::string format, TimeUnit::type unit)
 StrptimeOptions::StrptimeOptions() : StrptimeOptions("", TimeUnit::SECOND) {}
 constexpr char StrptimeOptions::kTypeName[];
 
+StrftimeOptions::StrftimeOptions(std::string format, std::string timezone)
+    : FunctionOptions(internal::kStrftimeOptionsType),
+      format(std::move(format)),
+      timezone(std::move(timezone)) {
+  tz = arrow_vendored::date::locate_zone(this->timezone);
+}
+StrftimeOptions::StrftimeOptions() : StrftimeOptions("%Y-%m-%dT%H:%M:%S%z", "UTC") {}
```

Comment:
The default is set to UTC, but for timestamp with timezone input, can we default to using the type's timezone?

Also for timestamp without timezone, it should probably not use `%z` by default?

And see my other comment, I suppose the default should rather be a literal `Z` instead of `%z` (or `%Z`) ?

Reply:
Switched to `Z` and removed `options.timezone` parameter to default  to type's timezone.